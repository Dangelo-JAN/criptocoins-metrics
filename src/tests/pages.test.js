import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
// import { fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';

import HomePage from '../pages/Home';
import DetailsPage from '../pages/Details';

describe('Pages renders correctly', () => {
  it('Home page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <HomePage />
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    );
    const element = document.querySelector('.coins-list');
    expect(element).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
  it('Details page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DetailsPage />
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <DetailsPage />
      </Provider>,
    );
    const element = document.querySelector('.coin-detail-container');
    expect(element).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});
