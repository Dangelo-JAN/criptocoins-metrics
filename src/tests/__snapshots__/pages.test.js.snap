// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages renders correctly Details page renders correctly 1`] = `
Array [
  <h1>
    ALL COIN STATS
  </h1>,
  <div
    className="container-column"
  >
    <div
      className="coin-detail-container"
    >
      <img
        className="coin-img"
      />
      <div
        className="container-column"
      >
        <div>
          <span
            className="title-details"
          >
            Name:
          </span>
          <span
            className="data"
          />
        </div>
        <div>
          <span
            className="title-details"
          >
            Symbol:
          </span>
          <span
            className="data"
          />
        </div>
        <div>
          <span
            className="title-details"
          >
            Ranking:
          </span>
          <span
            className="data"
          />
        </div>
      </div>
    </div>
    <div
      className="stats-container"
    >
      <span
        className="title-details"
      >
        Price:
      </span>
      <span
        className="data"
      />
      <span
        className="title-details"
      >
        Change %:
      </span>
      <span
        className="price-down"
      />
    </div>
  </div>,
]
`;

exports[`Pages renders correctly Home page renders correctly 1`] = `
Array [
  <input
    onChange={[Function]}
    placeholder="Search Coin"
    type="text"
  />,
  <ul
    className="container coins-list"
  />,
]
`;
